HttpServer.cs

```diff
@@ -154,22 +159,40 @@ class HttpServer {
 
     filename = Path.Combine(_rootDirectory, filename);
 
-    if (File.Exists(filename)) {
-      try {
-        Stream input = new FileStream(filename, FileMode.Open);
-
-        //Adding permanent http response headers
-        string mime;
-        context.Response.ContentType = _mimeTypeMappings.TryGetValue(Path.GetExtension(filename), out mime) ? mime : "application/octet-stream";
-        context.Response.ContentLength64 = input.Length;
-        context.Response.AddHeader("Date", DateTime.Now.ToString("r"));
-        context.Response.AddHeader("Last-Modified", System.IO.File.GetLastWriteTime(filename).ToString("r"));
+    bool isValidPath;
+    try {
+      isValidPath = new Uri(_rootDirectory).IsBaseOf(new Uri(filename));
+    }
+    catch (Exception) {
+      isValidPath = false;
+    }
```

Comment:
For this part, I think we could make it into a functions, so we coudl make it to be like this

bool checkIfValid () {
try{
return ...
}catch(err){
return ...
}
}